Core/HLE/sceUsbCam.cpp

```diff
@@ -89,65 +126,43 @@ static int sceUsbCamReadMicBlocking(u32 bufAddr, u32 size) {
 			Memory::Write_U8(i & 0xFF, bufAddr + i);
 		}
 	}
-	hleEatMicro(1000000 / micParam.frequency * (size / 2));
+	hleEatMicro(1000000 / config->micParam.frequency * (size / 2));
 	return size;
 }
 
 static int sceUsbCamSetupVideo(u32 paramAddr, u32 workareaAddr, int wasize) {
-	INFO_LOG(HLE, "UNIMPL sceUsbCamSetupVideo");
-	if (Memory::IsValidRange(paramAddr, sizeof(videoParam))) {
-		Memory::ReadStruct(paramAddr, &videoParam);
+	if (Memory::IsValidRange(paramAddr, sizeof(PspUsbCamSetupVideoParam))) {
+		Memory::ReadStruct(paramAddr, &config->videoParam);
 	}
-
-	INFO_LOG(HLE, "UNIMPL sceUsbCamSetupVideo - size: %d", videoParam.size);
-	INFO_LOG(HLE, "UNIMPL sceUsbCamSetupVideo - resolution: %d", videoParam.resolution);
-	INFO_LOG(HLE, "UNIMPL sceUsbCamSetupVideo - framesize: %d", videoParam.framesize);
-
-	std::lock_guard<std::mutex> lock(videoBufferMutex);
-	videoBufferLength = sizeof(sceUsbCamDummyImage);
-	memset(videoBuffer, 0, VIDEO_BUFFER_SIZE);
-	memcpy(videoBuffer, sceUsbCamDummyImage, sizeof(sceUsbCamDummyImage));
+	config->type = Camera::ConfigType::CfVideo;
 	return 0;
 }
 
-static int sceUsbCamStartVideo() {
-	INFO_LOG(HLE, "UNIMPL sceUsbCamStartVideo");
-#if defined(_WIN32) && !PPSSPP_PLATFORM(UWP)
-	if (winCamera) {
-		if (winCamera->isShutDown()) {
-			delete winCamera;
-			winCamera = new WindowsCaptureDevice(CAPTUREDEVIDE_TYPE::VIDEO);
-			winCamera->sendMessage({ CAPTUREDEVIDE_COMMAND::INITIALIZE, nullptr });
-		}
-
-		winCamera->sendMessage({ CAPTUREDEVIDE_COMMAND::START, nullptr });
+static int sceUsbCamSetupVideoEx(u32 paramAddr, u32 workareaAddr, int wasize) {
+	if (Memory::IsValidRange(paramAddr, sizeof(PspUsbCamSetupVideoExParam))) {
+		Memory::ReadStruct(paramAddr, &config->videoExParam);
 	}
-
-#else
-	System_SendMessage("camera_command", "startVideo");
-#endif
+	config->type = Camera::ConfigType::CfVideoEx;
 	return 0;
 }
 
-static int sceUsbCamStopVideo() {
-	INFO_LOG(HLE, "UNIMPL sceUsbCamStopVideo");
-#if defined(_WIN32) && !PPSSPP_PLATFORM(UWP)
-	if (winCamera)
-		winCamera->sendMessage({ CAPTUREDEVIDE_COMMAND::STOP, nullptr });
-#else
-	System_SendMessage("camera_command", "stopVideo");
-#endif
+static int sceUsbCamStartVideo() {
+	std::lock_guard<std::mutex> lock(videoBufferMutex);
+	videoBufferLength = sizeof(sceUsbCamDummyImage);
+	memset(videoBuffer, 0, VIDEO_BUFFER_SIZE);
+	memcpy(videoBuffer, sceUsbCamDummyImage, sizeof(sceUsbCamDummyImage));
+	Camera::startCapture();
 	return 0;
 }
 
-static int sceUsbCamAutoImageReverseSW(int rev) {
-	INFO_LOG(HLE, "UNIMPL sceUsbCamAutoImageReverseSW");
+static int sceUsbCamStopVideo() {
+	Camera::stopCapture();
 	return 0;
 }
 
 static int sceUsbCamReadVideoFrameBlocking(u32 bufAddr, u32 size) {
 	std::lock_guard<std::mutex> lock(videoBufferMutex);
-	u32 transferSize = std::min(videoBufferLength, size);
+	u32 transferSize = (std::min)(videoBufferLength, size);
```

Comment:
What's with the weird parentheses?

Reply:
https://stackoverflow.com/questions/5004858/why-is-stdmin-failing-when-windows-h-is-included . I replaced it with NOMINMAX